src/common/BaseLightAccount.sol

```diff
@@ -75,6 +76,39 @@ abstract contract BaseLightAccount is BaseAccount, TokenCallbackHandler, UUPSUpg
         }
     }
 
+    /// @notice Creates a contract, this can only be called by this account.
+    /// @param initCode The initCode to deploy. NOTE: This could be replaced with transient storage in the near future,
+    /// depending on gas savings, if any.
+    function create(bytes calldata initCode, uint256 value) external payable virtual onlyAuthorized {
+        assembly ("memory-safe") {
+            // Copy the initCode to memory, then deploy the contract
+            let len := initCode.length
+            calldatacopy(0, initCode.offset, len)
```

Comment:
Ah, technically this is not memory safe. We should either not mark the block as memory-safe and document this, or copy the `initCode` to a point past the free mem ptr.

While this probably works with compiling the code in its exact current state, there are some ways this can cause issues:
- If someone inherits from this contract, overwrites this virtual function with a new one, and calls `super.create`, then any memory vars they have allocated could be overwritten. This includes stack vars lifted to memory with `via-ir`, since those get allocated starting at `0x80`.
- If someone forks this and makes `create` public, calling it internally can do a similar thing.

Reply:
Agreed on the memory-safety, despite stepping through the function successfully, it makes sense to update everything to *actually* be memory-safe, even if execution halts, if only to save us future headache with the IR optimizer.